numpy/core/tests/test_longdouble.py

```diff
@@ -80,6 +81,11 @@ def test_fromstring():
                  err_msg="reading '%s'" % s)
 
 
+@in_foreign_locale
+def test_fromstring_best_effort():
+    assert_equal(np.fromstring("1,234", dtype=float, sep=" "),
```

Comment:
That's how fromstring normally "handles" invalid input. But if , is the decimal marker in your locale and you're in python 2.6, as @pv points out, you don't get the 1 either.